site/docs/features/management.md

```diff
@@ -1,177 +1,276 @@
+# You can inspect the files to be deleted this way ...
+nessie-gc list-deferred --live-set-id 0baaa1ff-90db-4ee5-b6d2-b60aea148c76 \
+  --jdbc... # JDBC settings omitted in this example
+
+# ... or this way
+nessie-gc show --live-set-id 0baaa1ff-90db-4ee5-b6d2-b60aea148c76 \
+  --with-deferred-deletes \
+  --jdbc... # JDBC settings omitted in this example
+
+# Now perform the file deletions
+nessie-gc deferred-deletes --live-set-id 0baaa1ff-90db-4ee5-b6d2-b60aea148c76 \
+  --jdbc... # JDBC settings omitted in this example
+```
+
+!!! note
+The _sweep_ phase does not access Nessie. It does use Iceberg and accesses the data lake.
+If _deferred deletion_ is requested, no files will be deleted.
 
 !!! note
-    Compaction will show up as a commit, like any other table operation.
+Since data lakes can easily contain a huge amount of files, the _expire_ phase does not remember
+every live data file (see the Iceberg assets above) individually, but uses a probabilistic data
+structure (bloom filer). The default settings expect, for each content ID, 1,000,000 files and
+uses a false-positive-probability of 0.0001 (those defaults may change, but can be inspected
+with `nessie-gc help expire`). The _expire_ phase will abort, if it hits a content-ID that
+_massively_ exceeds the configured false-positive-probability, because it hits way more live
+file references.
+
+!!! note
+Nessie GC's _expire_ phase processes up to 4 content-IDs in parallel. This setting can be
+changed using the `--expiry-parallelism` command line option.
+
+### Recommended production setup for Nessie GC
+
+It is highly recommended to use a Postgres or compatible or H2 database to persist the
+live-content-sets. Running the different Nessie GC phases separately is only supported with such a
+database.
+
+Make yourself familiar with all the commands offered by `nessie-gc` and the available command line
+options. It is safe to run `nessie-gc mark-live`, because it is non-destructive. Use
+`nessie-gc show --with-content-references` to inspect the collected live content references.
+
+Use _deferred deletion_ and inspect the files to be deleted **before** running
+`nessie-gc deferred-deletes`.
+
+Use separate invocations for the _mark_, the _sweep_ and the _deferred deletion_ phases.
+
+### All-in-one
+
+As briefly mentioned above, the `nessie-gc gc` command can be used to combine the _mark_ and _sweep_
+phases, optionally using the `--inmemory` option.
+
+`gc` is equivalent to first running `identify` and then `expire`, and it takes the same set of
+command line options. 
+
+### Troubleshooting
+
+Nessie GC tool emits the log output at `INFO` level. The default log level for the console can be
+overridden using the Java system property `log.level.console`, for example using the following
+command. So instead of directly running `nessie-gc`, just run it as an executable jar.
+
+```shell
+java -Dlog.level.console=DEBUG -jar $(which nessie-gc) 
+```
```

Comment:
Do we need to add a documentation with API level examples? (along with current tool/CLI level example). Because in the mailing list, some users has shown interest to use GC via API.

Reply:
At this point, the GC code is not a public & stable API.